### PR TITLE
Support ubx cfg pm2 message

### PR DESCRIPTION
**Pull request by nabelekt**
Adding support for UBX-CFG-PM2 messages

### CODE REVIEW

File: src/u-blox_structs.h [hunk 1]
```diff
@@ -1206,6 +1228,42 @@ typedef struct
   UBX_CFG_RATE_data_t  *callbackData;
 } UBX_CFG_RATE_t;
 
+
```
Comment (nabelekt) [inline src/u-blox_structs.h:1231]: Moving this up to be under the `// CFG-specific structs` section.

**Comment by PaulZC** [comment]
Hi Thomas ( @nabelekt ),

Thank you for submitting this PR.

I will not be able to merge it as it does not follow the structure and style of the existing library. Also there are some code changes you will need to make.

You use ```Serial.println``` in several places. Those need to be changed to:

```
    if (_printDebug == true)
    {
      _debugSerial->println(F("getPowerManagementConfiguration: PM2 configuration:"));
    }
```

```Serial.println(payloadCfg[])``` will print decimal values which is really not very useful. Please change those to something which indicates how the flags and updatePeriod etc. are configured.

Your typedef struct for UBX_CFG_PM2_data_t does not follow the structure and style of the other message structure definitions. Please change it so it matches the structure used by (e.g.) UBX_CFG_TP5_data_t.

Your code will need to support all three versions of the UBX-CFG-PM2 message - otherwise modules with different protocol versions will reject the message. You will need to use a read-modify-write approach. First get (poll) the PM2 message. Then look at the message version. Then update the fields and flags according to the protocol version. Then write it back again. Also, modules like the ZED-F9P do not support UBX_CFG_PM2, so your code will need to cope with the get (poll) being rejected.

Thank you.

Best wishes,
Paul

**Comment by PaulZC** [comment]
Hi Thomas (@nabelekt ),

I've just released v2.0.18 of the library. It includes some important changes to let the code run correctly on ESP32 v2.0.1. I have included some parts from this pull request. Please see the [release notes](https://github.com/sparkfun/SparkFun_u-blox_GNSS_Arduino_Library/releases/tag/v2.0.18) for more details. Please do pull the latest code and use that for the rest of your changes.

Please also see [CONTRIBUTING.md](https://github.com/sparkfun/SparkFun_u-blox_GNSS_Arduino_Library/blob/main/CONTRIBUTING.md). If possible, please target your pull request at the _release_candidate_ branch instead of the _main_ branch. That way, I can test your changes before merging them into the main branch.

Thank you.

Best wishes,
Paul

**Comment by nabelekt** [comment]
Hi, @PaulZC.

Thanks for the information and comments. I have merged in your master branch and made some other updates. However, I haven't reworked the struct, supported other message versions, or handled the message being rejected. My understanding is that any module that supports PM2 will support version 0x01 of the message; that's why I implemented that version. Unfortunately, I don't have the capacity right now to complete your suggestions. I'd welcome anyone else to do that. If I am able to later on, I'll come back to do that.

Thanks!

**Comment by PaulZC** [comment]
Hi Thomas (@nabelekt ),

Please be aware that I have just released v2.1.0 of the library. It adds a lot of new code to support AssistNow Online, Offline and Autonomous. Please do merge the new main branch when you have time.

Best wishes,
Paul

**Comment by PaulZC** [comment]
Hi Thomas,

I am going to close this PR. Please open a new pull request when you are ready to submit your code changes. Thank you.

Best wishes,
Paul

**Pull request closed without merging**